Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -3011,6 +3011,41 @@ internal static bool ShouldDrawWarningIconForBinding(string bindingPath)
 
 #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
 
+#if UNITY_EDITOR
+        [InitializeOnLoad]
+        public static class PlayStateNotifier
+        {
+            static PlayStateNotifier()
+            {
+                EditorApplication.playModeStateChanged += ModeChanged;
+            }
+
+            static void ModeChanged(PlayModeStateChange playModeState)
```

Comment:
If not already checked it would be good to investigate and document along with the already added inline docs (which is great by the way) when the actions may be expected to be enabled. E.g.

Are they enabled when entering Awake() on a MonoBehavior?
Are they enabled when entering Start() on a MonoBehavior?
Are they enabled when entering OnEnable() on a MonoBehavior?
Are they disabled when entering OnDisable() on a MonoBehavior?
Are they disabled when entering OnDestroy() on a MonoBehavior? (When destroy by exiting play mode)

Reply:
I have added an explanatory to the source (see comment I made to @lyndon-unity [below](https://github.com/Unity-Technologies/InputSystem/pull/1863#discussion_r1508913457))

Reply:
Awesome, thanks, that is helpful